src/app/pages/CourseDetail/CourseDetail.tsx

```diff
@@ -20,6 +21,21 @@ const priorities: Priorities = {
 function CourseDetail(): JSX.Element {
   const query = useQuery();
   const priority = query.get("priority");
+  const courseName = query.get("coursename");
+  console.log(courseName);
+
+  const [courses, setCourses] = useState<CourseData[]>([]);
+  useEffect(() => {
+    fetch("/api/coursedata")
```

Comment:
Instead of fetching all courses and filter them on client side, you could directly fetch the course your need.
Maybe with:
```ts
fetch(`/api/courses/${courseName}`)
```